packages/orchestration/src/exos/cosmos-orchestration-account.js

```diff
@@ -273,6 +348,124 @@ export const prepareCosmosOrchestrationAccountKit = (
+          if (!unbond) throw Fail`Result lacked unbond key: ${result}`;
+          return harden(unbond);
+        },
+      },
+      unbondingDelegationsQueryWatcher: {
+        /**
+         * @param {JsonSafe<ResponseQuery>[]} results
+         */
+        onFulfilled([result]) {
+          if (!result?.key) throw Fail`Error parsing result ${result}`;
+          const { unbondingResponses } =
+            QueryDelegatorUnbondingDelegationsResponse.decode(
+              decodeBase64(result.key),
+            );
+          if (!unbondingResponses)
+            throw Fail`Result lacked unbondingResponses key: ${result}`;
+          return harden(unbondingResponses);
+        },
+      },
+      redelegationQueryWatcher: {
+        /**
+         * @param {JsonSafe<ResponseQuery>[]} results
+         */
+        onFulfilled([result]) {
+          if (!result?.key) throw Fail`Error parsing result ${result}`;
+          const { redelegationResponses } = QueryRedelegationsResponse.decode(
+            decodeBase64(result.key),
+          );
+          if (!redelegationResponses)
+            throw Fail`Result lacked redelegationResponses key: ${result}`;
+          return harden(redelegationResponses);
+        },
+      },
+      redelegationsQueryWatcher: {
+        /**
+         * @param {JsonSafe<ResponseQuery>[]} results
+         */
+        onFulfilled([result]) {
+          if (!result?.key) throw Fail`Error parsing result ${result}`;
+          const { redelegationResponses } = QueryRedelegationsResponse.decode(
+            decodeBase64(result.key),
+          );
+          if (!redelegationResponses)
+            throw Fail`Result lacked redelegationResponses key: ${result}`;
+          return harden(redelegationResponses);
+        },
+      },
+      rewardQueryWatcher: {
+        /**
+         * @param {JsonSafe<ResponseQuery>[]} results
+         */
+        onFulfilled([result]) {
+          if (!result?.key) throw Fail`Error parsing result ${result}`;
+          const { rewards } = QueryDelegationRewardsResponse.decode(
+            decodeBase64(result.key),
+          );
+          if (!rewards) throw Fail`Result lacked rewards key: ${result}`;
+          return harden(rewards.map(toDenomAmount));
+        },
+      },
+      rewardsQueryWatcher: {
+        /**
+         * @param {JsonSafe<ResponseQuery>[]} results
+         * @returns {OrchestrationQueryDelegationTotalRewardsResponse}
+         */
+        onFulfilled([result]) {
+          if (!result?.key) throw Fail`Error parsing result ${result}`;
+          const { rewards, total } = QueryDelegationTotalRewardsResponse.decode(
+            decodeBase64(result.key),
+          );
+          if (!rewards || !total)
+            throw Fail`Result lacked rewards or total key: ${result}`;
+          return harden({
+            rewards: rewards.map(reward => ({
+              validatorAddress: reward.validatorAddress,
+              reward: reward.reward.map(toDenomAmount),
```

Comment:
Should we need to add logic to `toDenomAmount` to round down to the nearest whole integer? I see that this query returns `DecCoin` instead of the typical `Coin`.

Alternatively, accounting for decimals with ratios is an option, but I'm not sure it's worth the cost here.

Reply:
Good catch! I've added tests and helpers for this